docs/quickstarts/astro.mdx

```diff
@@ -0,0 +1,176 @@
+---
+title: Use Clerk with Astro
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your Astro application.
+---
+
+# Use Clerk with Astro
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your Astro application.
+
+<Steps>
+### Install `@clerk/astro`
+
+Clerk's Astro SDK has prebuilt components and helpers to make user authentication easier.
+
+To get started using Clerk with Astro, add the SDK to your project:
+
+<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+  ```bash filename="terminal"
+  npm install @astro/clerk
+  ```
+
+  ```bash filename="terminal"
+  yarn add @astro/clerk
+  ```
+
+  ```bash filename="terminal"
+  pnpm add @astro/clerk
+  ```
+</CodeBlockTabs>
+
+### Set environment variables
+
+Below is an example of your `.env` file.
+
+**Pro tip!** If you are signed into your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can find your keys in the Clerk Dashboard on the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page.
+
+<InjectKeys>
+
+```sh filename=".env"
+PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}
+CLERK_SECRET_KEY={{secret}}
+```
+
+</InjectKeys>
+
+### Update `.env.d.ts`
+
+Inside your `src/` directory, update the `.env.d.ts` file to ensure that the locals added by the middleware are properly typed.
+
+```ts filename="src/.env.d.ts" {3}
+/// <reference path="../.astro/types.d.ts" />
```

Comment:
that's a nice catch, but i actually don't know, do you think we should remove this line ?

Reply:
I would say to double check with how you or Rob created your Astro app (maybe its conditional?), but if its isn't part of a reasonable, common creation of a new Astro app and doesn't affect Clerk w/ Astro, probably better to remove it. It could cause confusion for newer developers.

Reply:
addressed this in a new [PR](https://github.com/clerk/clerk-docs/pull/1276), IDK why I added it lol, probably when doing some local tests. But it is 💯 not in a fresh astro app